cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -548,6 +564,34 @@ private static void validateConfiguration(Reconciliation reconciliation, Kafka k
         }
     }
 
+    /**
+     * Validates the user configuration with the configuration of quotas plugin
+     * In case that user configured the client.quota.callback.class option and the {@link QuotasPluginStrimzi} is configured as well,
+     * the warning is raised and the option is removed
+     *
+     * @param configuration     {@link KafkaConfiguration} with user specified config
+     * @param quotasPlugin      configuration of the quotas plugin
+     * @param warnings          list of warnings
+     */
+    private static void validateConfigurationOfQuotasPlugin(KafkaConfiguration configuration, QuotasPlugin quotasPlugin, List<Condition> warnings) {
+        if (quotasPlugin != null) {
+            if (quotasPlugin instanceof QuotasPluginStrimzi quotasPluginStrimzi) {
+                if (quotasPluginStrimzi.getMinAvailableBytesPerVolume() != null && quotasPluginStrimzi.getMinAvailableRatioPerVolume() != null) {
+                    throw new InvalidResourceException("You cannot configure both `minAvailableBytesPerVolume` and `minAvailableRatioPerVolume`, they are mutually exclusive.");
+                }
+
+                if (configuration.getConfigOption(CLIENT_CALLBACK_CLASS_OPTION) != null) {
+                    warnings.add(StatusUtils.buildWarningCondition("QuotasPluginConflict",
+                        String.format("Quotas plugin class cannot be configured in .spec.kafka.config, " +
+                            "when .spec.kafka.quotas contains configuration of `%s` plugin. " +
+                            "The plugin from .spec.kafka.quotas will be used", QuotasPluginStrimzi.TYPE_STRIMZI)));
+
+                    configuration.removeConfigOption(CLIENT_CALLBACK_CLASS_OPTION);
+                }
```

Comment:
We discussed this earlier and it was agreed that it should be checked only for the plugin mismatch in case of Strimzi quotas plugin. However, I agree that it should be checked no matter what plugin is configured there -> if Kafka or Strimzi. So I will change it to check and warn user when there will be quotas plugin already configured. Thanks